### ISSUE TITLE
ingest is truncating variable labels, dropping categories

### ISSUE THREAD

**Issue by donsizemore** [IQSS/dataverse#3982]
In testing Scholars' Portal's DDI Explorer we noticed missing data in DDIs pulled from a test server running 4.7. In troubleshooting, Jon pulled a 2008-vintage file from our production Dataverse and uploaded it to Harvard's production Dataverse (sorry).

Compare https://dataverse.unc.edu/api/datasets/export?exporter=ddi&persistentId=hdl%3A1902.29/H-2254
to https://dataverse.harvard.edu/api/datasets/export?exporter=ddi&persistentId=doi%3A10.7910/DVN/DRGMLC

Category labels seem to be inconsistently dropped. We've traced through the datatable, datavariable and variablecategory tables to find that the categories are not saved in the database, so we suspect a problem with ingest -- either timeout or memory. We've confirmed the problem exists with ingest of .sav and SPSS portable file formats.

I initially included 4.7 in the title, but we've reproduced this behavior on a test machine in 4.5.

**Comment by jonc1438** [comment]
I am not 100% sure but I think we have traced this to a data file problem. I have not been able to replicate this with a new/different POR file. We will still investigate but it looks like this is due to older Odum files from past repositories.

**Comment by pdurbin** [comment]
@jonc1438 thanks for the update. I see you and @donsizemore closed this issue but if there's anything you need, please let us know.

**Comment by donsizemore** [comment]
@pdurbin here's what we think happened: our older Harris files were originally converted from IBM EBCDIC format to ASCII prior to 1993. Jon likes to use Harris files for demos because he knows they have rich metadata, but we apparently picked a rash of problem files with long conversion histories when we were playing with DDI Explorer yesterday. I pulled a list of all '%spp-port%' original filetypes from our database and @jonc1438 spent the better part of the day spot-checking them, then we decided to close the ticket.

If @kcondon wants to beat up on Dataverse's ingest process for Q&A, may I suggest https://dataverse.unc.edu/dataset.xhtml?persistentId=hdl:1902.29/H-2254 or any of Odum's older Harris data. They originated in EBCDIC, were converted to ASCII, moved to a Spires database then onto Sun systems. The original paper copies are long gone, but if we find any Jon still has his hexadecimal slide rule.
